sylvia-derive/src/multitest.rs

```diff
@@ -606,98 +568,60 @@ impl<'a> ContractMtHelpers<'a> {
-        let exec_variants = MsgVariants::new(
-            source.as_variants(),
-            MsgType::Exec,
-            generic_params,
-            where_clause,
-        );
-        let query_variants = MsgVariants::new(
-            source.as_variants(),
-            MsgType::Query,
-            generic_params,
-            where_clause,
-        );
-        let sudo_variants = MsgVariants::new(
-            source.as_variants(),
-            MsgType::Sudo,
-            generic_params,
-            where_clause,
-        );
+        let exec_variants =
+            MsgVariants::new(source.as_variants(), MsgType::Exec, &[], where_clause);
+        let query_variants =
+            MsgVariants::new(source.as_variants(), MsgType::Query, &[], where_clause);
         let associated_error = source.items.iter().find_map(|item| match item {
-            ImplItem::Type(ty) if ty.ident == "Error" => Some(&ty.ty),
+            TraitItem::Type(ty) if ty.ident == "Error" => Some(&ty.ident),
             _ => None,
         });
+        let sudo_variants =
+            MsgVariants::new(source.as_variants(), MsgType::Sudo, &[], where_clause);
         let error_type: Type = match associated_error {
             Some(error) => parse_quote!(#error),
             // This should never happen as the `interface` macro requires the trait to have an associated `Error` type
             None => unreachable!(),
         };
 
-        let contract = &source.self_ty;
-        let contract_name = extract_contract_name(contract);
-
         Self {
             error_type,
             source,
-            generic_params,
+            associated_types,
             where_clause,
-            contract_name,
-            interfaces,
             exec_variants,
             query_variants,
             sudo_variants,
-            contract_module,
         }
     }
 
     pub fn emit(&self) -> TokenStream {
         let Self {
-            source,
             error_type,
-            interfaces,
-            generic_params,
+            source,
+            associated_types,
+            where_clause,
             exec_variants,
             query_variants,
             sudo_variants,
-            where_clause,
-            contract_module,
-            contract_name,
         } = self;
 
         let sylvia = crate_module();
 
-        let associated_types = ImplAssociatedTypes::new(source);
-        let interface_name = interface_name(self.source);
-        let trait_name = Ident::new(&format!("{}", interface_name), interface_name.span());
-
-        let module = interfaces
-            .get_only_interface()
-            .map(|interface| {
-                let module = &interface.module;
-                quote! { #module :: }
-            })
-            .unwrap_or(quote! {});
+        let interface_name = &source.ident;
+        let trait_name = Ident::new(&format!("{}Proxy", interface_name), interface_name.span());
```

Comment:
We should have common name for both Interface and Contract mt trait.

`Messages`, `Proxy`, we can discuss that.

Then we could rename `sylvia::multitest::Proxy` to `ContractProxy` as it is never imported to test module so we can save simpler name for the trait.